src/KOKKOS/fix_rigid_kokkos.cpp

```diff
@@ -0,0 +1,1835 @@
+    k_fflag.h_view(i,1) = fflag_buffer[3*i+1];
+    k_fflag.h_view(i,2) = fflag_buffer[3*i+2];
+  }
+
+  k_body.modify<LMPHostType>();
+  k_nrigid.modify<LMPHostType>();
+  k_tflag.modify<LMPHostType>();
+  k_fflag.modify<LMPHostType>();
+
+  k_body.sync<DeviceType>();
+  k_nrigid.sync<DeviceType>();
+  k_tflag.sync<DeviceType>();
+  k_fflag.sync<DeviceType>();
+
+  delete [] body_buffer;
+  delete [] nrigid_buffer;
+  delete [] tflag_buffer;
+  delete [] fflag_buffer;
+
+
+  if (debug_level && comm->me == 0) {
+    fprintf(screen, "Body contains:\n");
+    for (int i = 0; i < nmax; ++i) {
+      fprintf(screen, "%d ", body[i]);
+    }
+    fprintf(screen, "\n");
+  }
+
+
+}
+
+template <class DeviceType>
+FixRigidKokkos<DeviceType>::~FixRigidKokkos()
+{
+  memoryKK->destroy_kokkos(k_nrigid,nrigid);
+  memoryKK->destroy_kokkos(k_tflag,tflag);
+  memoryKK->destroy_kokkos(k_fflag,fflag);
+  memoryKK->destroy_kokkos(k_body,body);
+
+  memoryKK->destroy_kokkos(k_masstotal,masstotal);
+  memoryKK->destroy_kokkos(k_xcm,xcm);
+  memoryKK->destroy_kokkos(k_vcm,vcm);
+  memoryKK->destroy_kokkos(k_fcm,fcm);
+  memoryKK->destroy_kokkos(k_torque,torque);
+
+
+  memoryKK->destroy_kokkos(k_omega,omega);
+  memoryKK->destroy_kokkos(k_angmom,angmom);
+
+  memoryKK->destroy_kokkos(k_quat,quat);
+  memoryKK->destroy_kokkos(k_inertia,inertia);
+
+  memoryKK->destroy_kokkos(k_ex_space,ex_space);
+  memoryKK->destroy_kokkos(k_ey_space,ey_space);
+  memoryKK->destroy_kokkos(k_ez_space,ez_space);
+
+  memoryKK->destroy_kokkos(k_sum,sum);
+  memoryKK->destroy_kokkos(k_all,all);
+  memoryKK->destroy_kokkos(k_langextra,langextra);
+
+  memoryKK->destroy_kokkos(k_imagebody,imagebody);
+  memoryKK->destroy_kokkos(k_remapflag,remapflag);
+
+
+  memoryKK->destroy_kokkos(k_xcmimage, xcmimage);
+  memoryKK->destroy_kokkos(k_displace, displace);
+  memoryKK->destroy_kokkos(k_eflags,eflags);
+  memoryKK->destroy_kokkos(k_orient,orient);
+  memoryKK->destroy_kokkos(k_dorient,dorient);
+
+}
+
+
+
+
+
+template <class DeviceType>
+void FixRigidKokkos<DeviceType>::init()
+{
+  // The host code uses these in FixRigid::init():
+  // They should be synched first.
+  // tflag, fflag, body, mu, radius, rmass, mass, ellipsoid, line, tri, type,
+  // nlocal, eflags, image, x, sum, xcmimage, inertia, {ex,ey,ez}_space, quat
+  k_body.sync<LMPHostType>();
+  k_quat.sync<LMPHostType>();
+  k_inertia.sync<LMPHostType>();
+  k_ex_space.sync<LMPHostType>();
+  k_ey_space.sync<LMPHostType>();
+  k_ez_space.sync<LMPHostType>();
+
+  k_tflag.sync<LMPHostType>();
+  k_fflag.sync<LMPHostType>();
+  k_xcmimage.sync<LMPHostType>();
+  k_all.sync<LMPHostType>();
+  k_sum.sync<LMPHostType>();
+
+  k_vcm.sync<LMPHostType>();
+  k_angmom.sync<LMPHostType>();
+  k_imagebody.sync<LMPHostType>();
+  k_xcm.sync<LMPHostType>();
+  k_xcmimage.sync<LMPHostType>();
+  k_displace.sync<LMPHostType>();
+
+  atomKK->k_image.sync<LMPHostType>();
+  atomKK->k_x.sync<LMPHostType>();
```

Comment:
For views in the `atomKK` class like `image` and `x`, don't directly call `sync/modify`, use this wrapper call instead: `atomKK->sync(IMAGE_FLAG|X_FLAG);` I.e.   because the wrapper potentially adds more operations.

Reply:
Taken care of. Is there also a short-hand for syncing to the device?

Reply:
Haha, that does not actually compile. Did you mean atomKK->sync(LMPHostType(), IMAGE_MASK | X_MASK)?